### PR TITLE
feat: clean up synth.py by using IAM service option from generator

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**
Regenerate KMS client library using `IAM service` in gax-nodejs.
related feature: https://github.com/googleapis/gax-nodejs/pull/762

This feature helps us to remove the manual layer: iam_protos, iam_client_config.json, helper.ts, extra_proto_list.json, hacks in synth.py. This won't give any change to users, but only for automating the code generation.

### CODE REVIEW

File: src/v1/key_management_service_client.ts [hunk 1]
```diff
@@ -3415,23 +3416,23 @@ export class KeyManagementServiceClient {
    */
 
   getIamPolicy(
```
Comment (alexander-fenster) [inline src/v1/key_management_service_client.ts:3418]: We need to add jsdoc comments for these methods.

File: src/v1/key_management_service_client.ts [hunk 2]
```diff
@@ -3415,23 +3416,23 @@ export class KeyManagementServiceClient {
...
-    request: protos.google.iam.v1.GetIamPolicyRequest,
+    request: IamProtos.google.iam.v1.GetIamPolicyRequest,
     options: gax.CallOptions,
-    callback: protos.google.iam.v1.IAMPolicy.GetIamPolicyCallback
+    callback: IamProtos.google.iam.v1.IAMPolicy.GetIamPolicyCallback
   ) {
     return this._iamClient.getIamPolicy(request, options, callback);
   }
   setIamPolicy(
```
Comment (alexander-fenster) [inline src/v1/key_management_service_client.ts:3425]: I'm pretty sure all IAM methods support calling without a callback. Can we also have the declaration returning promises, as we do for regular simple methods?

Comment (xiaozhenliu-gg5) [reply src/v1/key_management_service_client.ts:3425]: sure, added at micro-gen side, thank you!

File: src/v1/key_management_service_client.ts (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply src/v1/key_management_service_client.ts:3418]: sounds great to me, let me check is there anydoc I can import for this.

Comment (xiaozhenliu-gg5) [reply src/v1/key_management_service_client.ts:3418]: stole it from original nodejs-kms repo. Thanks!

File: synth.py [hunk 3]
```diff
@@ -33,10 +33,11 @@
             'kms', version,
             generator_args={
                 "grpc-service-config": f"google/cloud/kms/{version}/cloudkms_grpc_service_config.json",
-                "package-name": f"@google-cloud/kms"
+                "package-name": f"@google-cloud/kms",
```
Comment (alexander-fenster) [inline synth.py:36]: nit: no need to have `f"  "` here (it's similar to JavaScript backtick string), since you don't have any variables there.

**Comment by alexander-fenster** [comment]
This should make you happier @JustinBeckwith - just look at those `synth.py` changes :)

File: synth.py (hunk 3 above)
Comment (xiaozhenliu-gg5) [reply synth.py:36]: cool! removed it! Thank you.

**Pull request merged**